scripts/expX2.py

```diff
@@ -0,0 +1,763 @@
+
+    for sut in SUTS:
+
+        state.sutsLeft -= 1
+        state.resetTmpForNewRun()
+
+        code = ""
+        completedForSut = 0
+
+        for seed in range(MIN_SEED, MAX_SEED + 1):
+
+            random.shuffle(CONFIGS)
+
+            for config in CONFIGS:
+
+                for setting in config.genAllSettings():
+
+                    # first run in current script: we need to create all the initializing preambles
+                    if state.counter == 0:
+                        code = createOneJob(state, sut, seed, setting)
+
+                    # can we add this new run to the current opened script?
+                    elif(
+                            # we need to check if we would not exceed the budget limit per job
+                            (state.counter + sut.timeWeight) <= state.perJob
+                            # however, that check must be ignored if we cannot open/create any new script file
+                            # for the current SUT
+                            or not state.hasSpareJobs() or
+                            # this case is bit more tricky... let's say only few runs are left that
+                            # we need to allocate in a script, but they are so few that they would need
+                            # only a small percentage of a new script capacity (eg, less than 30%).
+                            # In such a case, to avoid getting very imbalanced execution times,
+                            # we could just add those few runs to the current script.
+                            (NRUNS_PER_SUT - completedForSut < 0.3 * state.perJob / sut.timeWeight)
+                         ):
+                        code += addJobBody(state.port, sut, seed, setting)
+                        state.updateBudget(sut.timeWeight)
+
+                    else:
+                        writeWithHeadAndFooter(code, state.port, sut, state.getTimeoutMinutes())
+                        state.resetTmpForNewRun()
+                        code = createOneJob(state, sut, seed, setting)
+
+                    # keep track that a new run has been handled
+                    completedForSut += 1
+
+        if state.opened:
+            writeWithHeadAndFooter(code, state.port, sut, state.getTimeoutMinutes())
+
+    print("Generated scripts: " + str(state.generated))
+    print("Max wait for a job: " + str(max(state.waits)) + " minutes")
+    print("Median wait for a job: " + str(statistics.median(state.waits)) + " minutes")
+    print("Budget left: " + str(state.budget))
+    print("Total time: " + str(sum(state.waits) / 60) + " hours")
+    print("Total budget: " + str(CPUS * sum(state.waits) / 60) + " hours")
+
+
+
+############################################################################
+### Custom
+### Following will need to be changed based on what kind of experiments
+### we want to run.
+############################################################################
+
```

Comment:
based on the comment above, are the following (eg `ParameterSetting`) something the user should change?
or rather the comment `Custom...` should be moved down this file?